### PR TITLE
Add Chennai region and zones and update SysTypes

### PR DESCRIPTION
**Pull request by arshadd-b**
This PR adds regions and zones and  updates SysTypes
Fixes issue https://github.com/ppc64le-cloud/powervs-utils/issues/14

### CODE REVIEW

**Comment by mkumatag** [comment]
/cc @dharaneeshvrd @Karthik-K-N 

please review and also Chennai DC and also go through this doc - https://cloud.ibm.com/docs/power-iaas?topic=power-iaas-ibm-cloud-reg and see what else is missing.

**Comment by Prajyot-Parab** [comment]
@arshadd-b 
Add Chennai DC
```
zone   -> che01
region -> che
```

File: region.go [hunk 1]
```diff
@@ -100,11 +98,11 @@ var Regions = map[string]Region{
 	},
 	"mon": {
 		Description: "Montreal, Canada",
-		VPCRegion:   "",
+		VPCRegion:   "ca-tor",
```
Comment (Prajyot-Parab) [inline region.go]:
```suggestion
		VPCRegion:   "",
```

File: region.go [hunk 2]
```diff
@@ -100,11 +98,11 @@ var Regions = map[string]Region{
...
 		COSRegion:   "ca-tor",
 		Zones:       []string{"mon01"},
 		SysTypes:    []string{"s922", "e980"},
-		VPCZones:    []string{},
+		VPCZones:    []string{"ca-tor-1", "ca-tor-2", "ca-tor-3"},
```
Comment (Prajyot-Parab) [inline region.go]:
```suggestion
		VPCZones:    []string{},
```

File: region.go (hunk 1 above)
Comment (Prajyot-Parab) [reply region.go]: AFAIK Montreal does not support VPC

File: region.go (hunk 2 above)
Comment (Prajyot-Parab) [reply region.go]: AFAIK Montreal does not support VPC

**Review by Prajyot-Parab — CHANGES_REQUESTED** [review]

File: region.go [hunk 3]
```diff
@@ -11,13 +11,11 @@ func GetRegion(zone string) (region string, err error) {
 	case strings.HasPrefix(zone, "us-south"):
 		region = "us-south"
 	case strings.HasPrefix(zone, "dal"):
-		region = "dal"
+		region = "us-south"
```
Comment (dharaneeshvrd) [inline region.go]: Why do we need this change? Not sure it might cause some breakage to existing modules using this func. Please validate this across OpenShift and capi projects for its usage.

File: region.go [hunk 4]
```diff
@@ -181,6 +181,16 @@ var Regions = map[string]Region{
...
 		VPCZones: []string{"us-east-1", "us-east-2", "us-east-3"},
 	},
+	"che": {
+		Description: "Chennai, India",
+		VPCRegion:   "",
+		COSRegion:   "",
+		Zones: []string{
+			"che01",
+		},
+		SysTypes: []string{"s922", "e980"},
```
Comment (dharaneeshvrd) [inline region.go:191]: Hope you have validated the sys types available in Chennai! If not you can check via PowerVS workspace in cloud.

File: region.go (hunk 3 above)
Comment (arshadd-b) [reply region.go]: I had referred this mapping https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/blob/main/docs/book/src/reference/regions-zones-mapping.md
Could see that the zones `us-south
dal10
dal12` having same region `us-south`
same case with `wdc`

It is getting used just here , I don't think will create any issue
https://github.com/openshift/machine-api-provider-powervs/blob/e253fa30d99bb7ed00c1f27ca3d810fa682a1901/pkg/client/powervs_client.go#L162C36-L162C45

File: region.go (hunk 4 above)
Comment (arshadd-b) [reply region.go:191]: I have checked the machine types for this zone `che01 ` I can see only these two machine types 

Do we need to include `Dedicated host types` as well  in sysTypes ?

**Comment by arshadd-b** [comment]
I have noticed new machine Types added in zones, I have added those as well
But in some cases like in Zone `dal12 ` having machine types `E980, S922`and  `dal10 ` is having machine types `E980,
S1022, S922` . Machine Types are different for different zones. 
How should we handle these cases since we have single entry of zones belonging to same region in map ?

For now we have kept the common machine Types in zones
https://github.com/ppc64le-cloud/powervs-utils/blob/main/region.go#L65

File: region.go (hunk 3 above)
Comment (dharaneeshvrd) [reply region.go]: Ok I think @Karthik-K-N can comment here.

**Comment by dharaneeshvrd** [comment]
Individual zone based machine types would alter the current region map. 
If we are sure that region map is not directly consumed and only util funcs are consumed, then we can alter the map and util funcs.

**Comment by Karthik-K-N** [comment]
I dont have much idea on the impact, @hamzy @mjturek could you please take a look.

**Comment by mkumatag** [comment]
we shouldn't have made public if not required!
>Individual zone based machine types would alter the current region map.
If we are sure that region map is not directly consumed and only util funcs are consumed, then we can alter the map and util funcs.

**Comment by mkumatag** [comment]
> > Individual zone based machine types would alter the current region map.
> > If we are sure that region map is not directly consumed and only util funcs are consumed, then we can alter the map and util funcs.

File: region.go (hunk 3 above)
Comment (Karthik-K-N) [reply region.go]: I think when we are setting provider id we set <region>/<zone> so till now its dal/dal10, I think it will affect it.
@KeerthanaAP @Shilpa-Gokul whats the value you set for region when using dal10 zone? in IPI

Comment (KeerthanaAP) [reply region.go]: We set the powervs region as "dal" while using "dal10" zone and I think we use "us-south" for VPCRegion.
https://github.com/openshift/installer/blob/release-4.15/pkg/types/powervs/powervs_regions.go#L24C3-L24C6

**Comment by arshadd-b** [comment]
> I have noticed new machine Types added in zones, I have added those as well But in some cases like in Zone `dal12 ` having machine types `E980, S922`and `dal10 ` is having machine types `E980, S1022, S922` . Machine Types are different for different zones. How should we handle these cases since we have single entry of zones belonging to same region in map ?
> 
> For now we have kept the common machine Types in zones https://github.com/ppc64le-cloud/powervs-utils/blob/main/region.go#L65

Have created different issue for this https://github.com/ppc64le-cloud/powervs-utils/issues/16

**Comment by dharaneeshvrd** [comment]
Overall LGTM!
@arshadd-b Please update the commit message with proper details!

**Comment by arshadd-b** [comment]
> Overall LGTM! @arshadd-b Please update the commit message with proper details!

done

File: region.go [hunk 5]
```diff
@@ -16,8 +16,6 @@ func GetRegion(zone string) (region string, err error) {
 		region = "sao"
 	case strings.HasPrefix(zone, "us-east"):
 		region = "us-east"
-	case strings.HasPrefix(zone, "tor"):
```
Comment (mkumatag) [inline region.go:19]: why is this removed? I still see an entry here for tor - https://cloud.ibm.com/docs/power-iaas?topic=power-iaas-ibm-cloud-reg

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: region.go (hunk 5 above)
Comment (mkumatag) [reply region.go:19]: @Prajyot-Parab @dharaneeshvrd @Karthik-K-N am I missing something here?

Comment (dharaneeshvrd) [reply region.go:19]: Its a [dup entry](https://github.com/ppc64le-cloud/powervs-utils/pull/15/files#diff-f53390ca7a4e1f4498b2ff8c143307da643ff451b0180b733ed294a9dcd84bc0L37), that's why she removed I guess.

Comment (arshadd-b) [reply region.go:19]: Yeah two entries were there for `tor`, so I have removed one

**Review by mkumatag — APPROVED** [review]
/lgtm
